### ISSUE TITLE
Allow `tenant_id` to be templatable on `loki` sink

### ISSUE THREAD

**Issue by jszwedko** [timberio/vector#5202]
Requested by user in discord: https://discord.com/channels/742820443487993987/746070604192415834/780815488774832158

They'd like to use the config:

```toml
[sinks.out]
  # General
  endpoint = "http://XXXX"
  inputs = ["my_transform_id"]
  type = "loki"
  labels.key = "value" # example
  labels.key = "{{ event_field }}" # example
  tenant_id = "{{ kubernetes.namespace }}"
```

It looks like this will require partitioned batching as tenant_id is sent per request.

https://github.com/timberio/vector/blob/18b9c0fade267e4833fa62067ba061ca84f35801/src/sinks/loki.rs#L187-L189

**Comment by juchiast** [comment]
Will this also resolve #5201?

**Comment by binarylogic** [comment]
@juchiast yep. Will close that one in favor of this one since it is better described.

**Comment by juchiast** [comment]
@mrmassis I have merged #5204 , it is now possible to use template syntax in `tenant_id` in master branch.

**Comment by mrmassis** [comment]
Great work!
